server/dbutils.go

```diff
@@ -0,0 +1,130 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"log"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/interaction"
+	"github.com/joshprzybyszewski/cribbage/server/persistence"
+	"github.com/joshprzybyszewski/cribbage/server/play"
+)
+
+var (
+	errInvalidUsername error = errors.New(`invalid username`)
+)
+
+func commitOrRollback(db persistence.DB, err *error) {
+	var err2 error
+	if *err != nil {
+		err2 = db.Rollback()
+	} else {
+		err2 = db.Commit()
+	}
+	if err2 != nil {
+		log.Printf("Could not commit/rollback after %+v: %+v\n", err, err2)
+	}
+}
+
+func handleAction(_ context.Context, db persistence.DB, action model.PlayerAction) error {
+	err := db.Start()
+	if err != nil {
+		return err
+	}
+	defer commitOrRollback(db, &err)
+
+	g, err := db.GetGame(action.GameID)
+	if err != nil {
+		return err
+	}
+
+	pAPIs, err := getPlayerAPIs(db, g.Players)
+	if err != nil {
+		return err
+	}
+	err = play.HandleAction(&g, action, pAPIs)
+	if err != nil {
+		return err
+	}
+	return db.SaveGame(g)
+}
+
+func createGame(_ context.Context, db persistence.DB, pIDs []model.PlayerID) (model.Game, error) {
+	err := db.Start()
+	if err != nil {
+		return model.Game{}, err
+	}
+	defer commitOrRollback(db, &err)
+
+	players := make([]model.Player, len(pIDs))
+	for i, id := range pIDs {
+		p, err2 := db.GetPlayer(id)
+		err = err2
```

Comment:
Why re-assign `err2`? we should be able to avoid that

Reply:
`p, err := db.GetPlayer(id)` declares a new `err` scoped to the `for` which the linter complains about

Reply:
Oh - and don't we want to keep `err` up to date for `commitOrRollback`?

Reply:
🔮 Perhaps we could declare `var p model.Player` outside of the scope of the for loop? then `p` is only reserved/assigned once, and we declare `p, err = db.GetPlayer(id)` here...

Reply:
Yes, much better